File: 코딩테스트 입문/옹알이 (1).cpp
#include <string>
#include <vector>

using namespace std;

vector<string> v = {"aya", "ye", "woo", "ma"};

int solution(vector<string> babbling) {
    int answer = 0;

    for (string bab : babbling) {
        bool valid = true;
        for(int i = 0; i < bab.length(); i++) {
            if(bab.substr(i, 3) == "aya") i += 2;
            else if(bab.substr(i, 2) == "ye") i += 1;
            else if(bab.substr(i, 3) == "woo") i += 2;
            else if(bab.substr(i, 2) == "ma") i += 1;
            else {
                valid = false;
                break;
            }
        }
        if (valid) answer++;
    }

    return answer;
}

File: 코딩테스트 입문/옹알이 (1).java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;

        for (int i = 0; i < babbling.length; i++) {
            boolean valid = true;
            for(int j = 0; j < babbling[i].length(); j++) {
                if(j + 3 <= babbling[i].length() && babbling[i].substring(j, j + 3).equals("aya")) j += 2;
                else if(j + 2 <= babbling[i].length() && babbling[i].substring(j, j + 2).equals("ye")) j += 1;
                else if(j + 3 <= babbling[i].length() && babbling[i].substring(j, j + 3).equals("woo")) j += 2;
                else if(j + 2 <= babbling[i].length() && babbling[i].substring(j, j + 2).equals("ma")) j += 1;
                else {
                    valid = false;
                    break;
                }
            }
            if (valid) answer++;
        }

        return answer;
    }
}